+Prefs/Button.m

```diff
@@ -1,23 +1,23 @@
 classdef Button < Base.pref
     %BUTTON for access to a "set" method that the user can activate on click. For instance, 
     %
-    %    Prefs.Button('name', 'Greeting', 'string', 'Click Me!' 'set', @(~,~)(disp('Hello World')) )
+    %    Prefs.Button('Click Me!', 'name', 'Greeting', 'set', @(~,~)(disp('Hello World')) )
     %
     % will create a UI line with 'Greeting: [ Click Me! ]' where the button is square bracketed.
     % Clicking the button will execute the set function and display 'Hello World' in the console.
     
     properties(Hidden)
-        default = false;    % Completely unused.
+        default = '';
         ui = Prefs.Inputs.ButtonField;
-        string = {'', @(a)validateattributes(a,{'char'},{'vector'})};   % String to display on the button
     end
     
     methods
         function obj = Button(varargin)
             obj = obj@Base.pref(varargin{:});
         end
-        function val = clean(~, ~)
-            val = false;
+        function validate(obj,val)
+            val = obj.value;
+%             validateattributes(val,{'char','string'},{'scalartext'})
```

Comment:
This seems like this should not be commented out

Reply:
where are these pref validation functions called actually? or is this for future?

Reply:
This is a Michael thing.